File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/model/SourceModel.java
package io.qimia.uhrwerk.common.model;

import com.google.common.base.Objects;
import java.io.Serializable;

public class SourceModel extends StateModel implements Comparable<SourceModel>, Serializable {
  private static final long serialVersionUID = 407099634243598968L;

  Long tableId;
  Long connectionId;
  String path;
  String format;
  ConnectionModel connection;
  TableModel table;
  PartitionUnit partitionUnit;
  int partitionSize;
  String parallelLoadQuery;
  String parallelLoadColumn;
  int parallelLoadNum;
  String selectQuery;
  String selectColumn;
  boolean partitioned = false;
  boolean autoLoad = true;

  public SourceModel(SourceModelBuilder builder) {
    super(builder.deactivatedTs);
    this.id = builder.id;
    this.tableId = builder.tableId;
    this.connectionId = builder.connectionId;
    this.path = builder.path;
    this.format = builder.format;
    this.connection = builder.connection;
    this.table = builder.table;
    this.partitionUnit = builder.partitionUnit;
    this.partitionSize = builder.partitionSize;
    this.parallelLoadQuery = builder.parallelLoadQuery;
    this.parallelLoadColumn = builder.parallelLoadColumn;
    this.parallelLoadNum = builder.parallelLoadNum;
    this.selectQuery = builder.selectQuery;
    this.selectColumn = builder.selectColumn;
    this.partitioned = builder.partitioned;
    this.autoLoad = builder.autoLoad;
  }

  public static SourceModelBuilder builder() {
    return new SourceModelBuilder();
  }

  public Long getTableId() {
    return tableId;
  }

  public void setTableId(Long tableId) {
    this.tableId = tableId;
  }

  public Long getConnectionId() {
    return connectionId;
  }

  public void setConnectionId(Long connectionId) {
    this.connectionId = connectionId;
  }

  public ConnectionModel getConnection() {
    return connection;
  }

  public void setConnection(ConnectionModel connection) {
    this.connection = connection;
  }

  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }

  public String getFormat() {
    return format;
  }

  public void setFormat(String format) {
    this.format = format;
  }

  public PartitionUnit getPartitionUnit() {
    return partitionUnit;
  }

  public void setPartitionUnit(PartitionUnit partitionUnit) {
    this.partitionUnit = partitionUnit;
  }

  public int getPartitionSize() {
    return partitionSize;
  }

  public void setPartitionSize(int partitionSize) {
    this.partitionSize = partitionSize;
  }

  public String getParallelLoadQuery() {
    return parallelLoadQuery;
  }

  public void setParallelLoadQuery(String parallelLoadQuery) {
    this.parallelLoadQuery = parallelLoadQuery;
  }

  public String getParallelLoadColumn() {
    return parallelLoadColumn;
  }

  public void setParallelLoadColumn(String parallelLoadColumn) {
    this.parallelLoadColumn = parallelLoadColumn;
  }

  public int getParallelLoadNum() {
    return parallelLoadNum;
  }

  public void setParallelLoadNum(int parallelLoadNum) {
    this.parallelLoadNum = parallelLoadNum;
  }

  public String getSelectQuery() {
    return selectQuery;
  }

  public void setSelectQuery(String selectQuery) {
    this.selectQuery = selectQuery;
  }

  public String getSelectColumn() {
    return selectColumn;
  }

  public void setSelectColumn(String selectColumn) {
    this.selectColumn = selectColumn;
  }

  public boolean isPartitioned() {
    return partitioned;
  }

  public void setPartitioned(boolean partitioned) {
    this.partitioned = partitioned;
  }

  public boolean isAutoLoad() {
    return autoLoad;
  }

  public void setAutoLoad(boolean autoLoad) {
    this.autoLoad = autoLoad;
  }

  @Override
  public int compareTo(SourceModel o) {
    return this.getId().compareTo(o.getId());
  }

  public TableModel getTable() {
    return table;
  }

  public void setTable(TableModel table) {
    this.table = table;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof SourceModel)) {
      return false;
    }
    if (!super.equals(o)) {
      return false;
    }
    SourceModel that = (SourceModel) o;
    return getPartitionSize() == that.getPartitionSize()
        && getParallelLoadNum() == that.getParallelLoadNum()
        && isPartitioned() == that.isPartitioned()
        && isAutoLoad() == that.isAutoLoad()
        && Objects.equal(getTableId(), that.getTableId())
        && Objects.equal(getConnectionId(), that.getConnectionId())
        && Objects.equal(getPath(), that.getPath())
        && Objects.equal(getFormat(), that.getFormat())
        && getPartitionUnit() == that.getPartitionUnit()
        && Objects.equal(getParallelLoadQuery(), that.getParallelLoadQuery())
        && Objects.equal(getParallelLoadColumn(), that.getParallelLoadColumn())
        && Objects.equal(getSelectQuery(), that.getSelectQuery())
        && Objects.equal(getSelectColumn(), that.getSelectColumn());
  }

  @Override
  public int hashCode() {
    return Objects.hashCode(
        super.hashCode(),
        getTableId(),
        getConnectionId(),
        getPath(),
        getFormat(),
        getPartitionUnit(),
        getPartitionSize(),
        getParallelLoadQuery(),
        getParallelLoadColumn(),
        getParallelLoadNum(),
        getSelectQuery(),
        getSelectColumn(),
        isPartitioned(),
        isAutoLoad());
  }

  @Override
  public String toString() {
    return "SourceModel{" +
        "tableId=" + tableId +
        ", connectionId=" + connectionId +
        ", path='" + path + '\'' +
        ", format='" + format + '\'' +
        ", partitionUnit=" + partitionUnit +
        ", partitionSize=" + partitionSize +
        ", parallelLoadQuery='" + parallelLoadQuery + '\'' +
        ", parallelLoadColumn='" + parallelLoadColumn + '\'' +
        ", parallelLoadNum=" + parallelLoadNum +
        ", selectQuery='" + selectQuery + '\'' +
        ", selectColumn='" + selectColumn + '\'' +
        ", partitioned=" + partitioned +
        ", autoLoad=" + autoLoad +
        '}';
  }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/model/SourceModel.kt
package io.qimia.uhrwerk.common.metastore.model

import java.sql.Timestamp
import java.time.LocalDateTime

data class SourceModel(
    var id: Long? = null,
    var tableId: Long? = null,
    var connectionId: Long? = null,
    var path: String? = null,
    var format: String? = null,
    var connection: ConnectionModel? = null,
    var table: TableModel? = null,
    var partitionUnit: PartitionUnit? = null,
    var partitionSize: Int? = null,
    var parallelLoadQuery: String? = null,
    var parallelLoadColumn: String? = null,
    var parallelLoadNum: Int? = null,
    var selectQuery: String? = null,
    var selectColumn: String? = null,
    var partitioned: Boolean = false,
    var autoLoad: Boolean = true,
    var description: String? = null,
    var deactivatedTs: LocalDateTime? = null,
    var createdTs: Timestamp? = null,
    var updatedTs: Timestamp? = null
) : BaseModel {
    override fun id(id: Long?) {
        this.id = id
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is SourceModel) return false

        if (id != null && other.id != null) if (id != other.id) return false
        if (tableId != other.tableId) return false
        if (connectionId != other.connectionId) return false
        if (path != other.path) return false
        if (format != other.format) return false
        if (connection != other.connection) return false
        if (table != other.table) return false
        if (partitionUnit != other.partitionUnit) return false
        if (partitionSize != other.partitionSize) return false
        if (parallelLoadQuery != other.parallelLoadQuery) return false
        if (parallelLoadColumn != other.parallelLoadColumn) return false
        if (parallelLoadNum != other.parallelLoadNum) return false
        if (selectQuery != other.selectQuery) return false
        if (selectColumn != other.selectColumn) return false
        if (partitioned != other.partitioned) return false
        if (autoLoad != other.autoLoad) return false
        if (description != other.description) return false
        if (deactivatedTs != other.deactivatedTs) return false
        if (createdTs != other.createdTs) return false
        if (updatedTs != other.updatedTs) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (tableId?.hashCode() ?: 0)
        result = 31 * result + (connectionId?.hashCode() ?: 0)
        result = 31 * result + (path?.hashCode() ?: 0)
        result = 31 * result + (format?.hashCode() ?: 0)
        result = 31 * result + (connection?.hashCode() ?: 0)
        result = 31 * result + (table?.hashCode() ?: 0)
        result = 31 * result + (partitionUnit?.hashCode() ?: 0)
        result = 31 * result + (partitionSize ?: 0)
        result = 31 * result + (parallelLoadQuery?.hashCode() ?: 0)
        result = 31 * result + (parallelLoadColumn?.hashCode() ?: 0)
        result = 31 * result + (parallelLoadNum ?: 0)
        result = 31 * result + (selectQuery?.hashCode() ?: 0)
        result = 31 * result + (selectColumn?.hashCode() ?: 0)
        result = 31 * result + partitioned.hashCode()
        result = 31 * result + autoLoad.hashCode()
        result = 31 * result + (description?.hashCode() ?: 0)
        result = 31 * result + (deactivatedTs?.hashCode() ?: 0)
        result = 31 * result + (createdTs?.hashCode() ?: 0)
        result = 31 * result + (updatedTs?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "SourceModel(id=$id, tableId=$tableId, connectionId=$connectionId, path=$path, format=$format, partitionUnit=$partitionUnit, partitionSize=$partitionSize, parallelLoadQuery=$parallelLoadQuery, parallelLoadColumn=$parallelLoadColumn, parallelLoadNum=$parallelLoadNum, selectQuery=$selectQuery, selectColumn=$selectColumn, partitioned=$partitioned, autoLoad=$autoLoad, description=$description, deactivatedTs=$deactivatedTs, createdTs=$createdTs, updatedTs=$updatedTs)"
    }
}